src/Chip/index.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Icon from '../Icon';
+
+const TAG_VARIANTS = [
+  'button',
+  'a',
+];
+
+const STYLE_VARIANTS = [
+  'light',
+  'dark',
+];
+
+const Chip = ({
+  as,
+  children,
+  className,
+  variant,
+  active,
+  iconBefore,
+  iconAfter,
+  ...props
+}) => {
+  const elementType = TAG_VARIANTS.includes(as) ? as : 'button';
+  const elementProps = {
+    ...props,
+    className: classNames(
+      'pgn__chip',
+      `pgn__chip-${variant}`,
+      { active },
+      className,
+    ),
+  };
+  const elementContent = (
+    <>
+      {iconBefore && <Icon className={classNames('chip-icon-before')} src={iconBefore} />}
+      {children}
+      {iconAfter && <Icon className={classNames('chip-icon-after')} src={iconAfter} />}
+    </>
+  );
+  return React.createElement(props.href ? 'a' : elementType, elementProps, elementContent);
+};
+
+Chip.propTypes = {
+  /** Specifies the base HTML element. */
+  as: PropTypes.elementType,
+  /** Specifies an additional `className` to add to the base element. */
+  className: PropTypes.string,
+  /** Specifies the content of the `Chip`. */
+  children: PropTypes.node,
+  /** The `Chip` style variant to use. */
+  variant: PropTypes.oneOf(STYLE_VARIANTS),
+  /** Adds the `active` class to the `Chip`. */
+  active: PropTypes.bool,
+  /** Disables the `Chip`. */
+  disabled: PropTypes.bool,
+  /** Providing a `href` will render an `<a>` element, styled as a button. */
+  href: PropTypes.string,
+  /**
+   * An icon component to render before the content.
+   * Example import of a Paragon icon component:
+   *
+   * `import { Check } from '@edx/paragon/icons';`
+   */
+  iconBefore: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
```

Comment:
Why do you allow icons to be functions also? 🤔 
Shouldn't it just be `PropTypes.node`?

Reply:
I copied from the Button component, but yes it should be just `PropTypes.node`. Editing, thanks